solidity/contracts/domains/interfaces/IConfidentialToken.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+/**
+ * @title IConfidentialToken
+ * @dev Base interface for a confidential UTXO token.
+ */
+interface IConfidentialToken {
+    event Transfer(
+        bytes32 txId,
+        address indexed operator,
+        bytes32[] inputs,
+        bytes32[] outputs,
+        bytes proof,
+        bytes data
+    );
+
+    /**
+     * @dev Spend UTXOs and create new UTXOs.
+     *
+     * @param txId a unique identifier for this transaction which must not have been used before
+     * @param inputs array of zero or more UTXOs that the signer is authorized to spend
+     * @param outputs array of zero or more new UTXOs to generate, for future transactions to spend
+     * @param proof implementation-specific proof for this transaction - may be validated by
+     *              the smart contract, or may represent evidence of off-chain validation
+     * @param data any additional transaction data (opaque to the blockchain)
+     *
+     * Emits a {Transfer} event.
+     */
+    function transfer(
+        bytes32 txId,
+        bytes32[] calldata inputs,
```

Comment:
What about "read" and "info" states that are needed for other data not mutated.

Few examples:
- Allow-lists for ZKPs ("read")
- Account states for private EVM execution that are not modified ("read")
- Transaction input payload reference ("info")
- Transaction manifest for establishing subsets of states visible to each party ("info")